### PR TITLE
fix(Scripts/Brewfest): Fixed race for tokens dailies.

### LINKED ISSUE #13077: [Brewfest] Race for tokens daily not working
original report https://github.com/chromiecraft/chromiecraft/issues/4102. confirmed on ptr
### Current Behaviour

Clicking "i want to race for tokens" does not start the "there and back again!" daily quest for tokens

### Expected Blizzlike Behaviour

completion of "there and back again!" quest from the npc Neill Ramstein (https://wowgaming.altervista.org/aowow/?npc=23558) and Ram Master Ray (https://wowgaming.altervista.org/aowow/?npc=24497) should unlock a repeatable daily for extra tokens

### Source

https://www.wowhead.com/wotlk/guides/brewfest-event-achievements#repeatable-quests

### Steps to reproduce the problem

.go creature 88857
.quest add 11122
.quest complete 11122
click dialogue "I Want to race for token"

### Extra Notes

https://www.wowhead.com/wotlk/quest=11122/there-and-back-again
image

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/04db1914d839060501eacaa13b6fd1e5d274677c

### Operating system

Windows 11

### Custom changes or Modules

[mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
    [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
    [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
    [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
    [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
    [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
    [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
    [mod-dmf-switch](https://github.com/azerothcore/mod-dmf-switch)
    [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
    [mod-eluna](https://github.com/azerothcore/mod-eluna)
    [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
    [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
    [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
    [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
    [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
    [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
    [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
    [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
    [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
    [mod-transmog](https://github.com/azerothcore/mod-transmog)
    [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
    [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
    [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
    [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
    [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
    [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
    [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
    [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
    [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
    [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
    [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### PR DESCRIPTION
**Pull request by UltraNix**
Fixes #13077

## Issues Addressed:

- Closes #13077
- Closes https://github.com/chromiecraft/chromiecraft/issues/4102

## Tests Performed:

- Not tested.

## How to Test the Changes:

More info [here](#13077)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Events/brewfest.cpp [hunk 1]
```diff
@@ -112,30 +112,21 @@ struct npc_brewfest_keg_reciver : public ScriptedAI
         }
     }
 
-    bool OnGossipSelect(Player* player, Creature* /*creature*/, uint32 /*uiSender*/, uint32 uiAction)
+    void sGossipSelect(Player* player, uint32 uiSender, uint32 uiAction) override
     {
-        switch (uiAction)
+        if (uiSender == 8953 || uiSender == 8973)
```
Comment (Kitzunu) [inline src/server/scripts/Events/brewfest.cpp:117]: .

File: src/server/scripts/Events/brewfest.cpp [hunk 2]
```diff
@@ -112,30 +112,21 @@ struct npc_brewfest_keg_reciver : public ScriptedAI
...
-        if (!player->HasSpellCooldown(SPELL_COOLDOWN_CHECKER) && player->GetQuestRewardStatus(player->GetTeamId() == TEAM_ALLIANCE ? QUEST_THERE_AND_BACK_AGAIN_A : QUEST_THERE_AND_BACK_AGAIN_H))
-            AddGossipItemFor(player, GOSSIP_ICON_CHAT, "Do you have additional work?", GOSSIP_SENDER_MAIN, GOSSIP_ACTION_INFO_DEF + 1);
+        if (uiAction != 4)
+        {
+            CloseGossipMenuFor(player);
 
-        SendGossipMenuFor(player, (creature->GetEntry() == NPC_NEILL_RAMSTEIN ? 8934 : 8976), creature->GetGUID());
-        return true;
+            player->CastSpell(player, 43883, true);
+            player->CastSpell(player, 44262, true);
```
Comment (Kitzunu) [inline src/server/scripts/Events/brewfest.cpp:128]: .

**Review by Kitzunu — COMMENTED** [review]
You know what I feel about magic numbers :D

**Comment by Gultask** [comment]
Hmm. I'm not exactly sure how to test, but from what I've seen the "I want to race for tokens!" still does nothing. Clicking the option will also lock up the gossip screen, preventing the other options from having any effect at all.

`.quest add 11318`
`.quest complete 11318`
`.go creature 88857`
`.quest add 11122`
`.quest complete 11122`

turn in both quests, the option still does nothing.

**Comment by UltraNix** [comment]
@Gultask Yeah - I forgot to add  it. You should test this PR with #13108

**Comment by UltraNix** [comment]
@Kitzunu Done.

**Review by Kitzunu — APPROVED** [review]

**Comment by Gultask** [comment]
Testing with https://github.com/azerothcore/azerothcore-wotlk/pull/13108 I don't see the option anymore, but I believe it's because I'm missing some condition here for it to spawn :S 

Also, when I click the "Do you still need help shipping kegs from Kharanos?" it adds the Ram aura and everything. But if I cancel the aura and dismount, these auras stay, preventing me from selecting the option again, is that correct?
image

**Comment by UltraNix** [comment]
Yep - there is a hidden cooldown you can read about in the commend in wowhead.
The option is replaced with blizzlike `Do you still need help shipping kegs from Kharanos?` - it does the same.

**Comment by Gultask** [comment]
Omg sorry Nix. I didn't do my research. I thought it was an actual quest and not just a gossip option. It's all correct, and blizzlike from what I've seen according to WotLK Classic.

**Pull request merged**